src/arrayMethodSort.js

```diff
@@ -5,7 +5,57 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const array = [...this];
+
+    this.length = 0;
+
+    let compare = compareFunction;
+
+    if (!compareFunction) {
```

Comment:
It's a good practice to write helper functions, like this, outside the scope of your `applyCustomSort()`,

```
const compareAsFunction = (a, b) => {
// your code
}
```

Then:
` [].__proto__.sort2 = function(compareFunction = compareAsFunction) {
// your code
}`